Cron/Order.php

```diff
@@ -190,10 +190,21 @@ protected function processOrderStatusUpdate(OrderInterface $order, string $newSt
     protected function cancelOrder(OrderInterface $order) : bool
     {
         try {
+            $isUnholded = false;
+            if ($order->canUnhold()) {
+                $isUnholded = true;
+                $order->unhold();
+            }
+
             if (!$order->canCancel()) {
                 $this->loggingClient->info(
                     sprintf('Unable to cancel Order #%s as it cannot be canceled', $order->getIncrementId())
                 );
+
+                if ($isUnholded) {
+                    $order->hold();
```

Comment:
If we can't cancel, can we hold? Are we sure this is the desired behavior?

Reply:
@crfroehlich If we can't cancel and the order was originally holded then we need to revert it to the original state of the order so we don't break the Merchant's intended state - This is the desired behavior